libpromises/evalfunction.c

```diff
@@ -1331,79 +1349,159 @@ static FnCallResult FnCallPackagesMatching(ARG_UNUSED EvalContext *ctx, ARG_UNUS
         GetSoftwarePatchesFilename(filename);
     }
 
-    Log(LOG_LEVEL_DEBUG, "%s: reading inventory from '%s'", fp->name, filename);
-
+    Log(LOG_LEVEL_DEBUG, "Reading inventory from '%s'", filename);
+    
     FILE *const fin = fopen(filename, "r");
     if (fin == NULL)
     {
         Log(LOG_LEVEL_VERBOSE,
-            "%s cannot open the %s packages inventory '%s' - "
+            "Cannot open the %s packages inventory '%s' - "
             "This is not necessarily an error. "
             "Either the inventory policy has not been included, "
-            "or it has not had time to have an effect yet. "
+            "or it has not had time to have an effect yet or you are using"
+            "new package promise and check for legacy promise is made."
             "A future call may still succeed. (fopen: %s)",
-            fp->name,
             installed_mode ? "installed" : "available",
             filename,
             GetErrorStr());
 
-        pcre_free(matcher);
-        return FnFailure();
+        return true;
```

Comment:
This is similar to above. In some cases you can end up here (for example when new package promise is used and package module is not expanded yet - pre-evaluation). If we return ```false``` here we will see promise error logs which is not expected.